mhkit/tests/power/test_power.py

```diff
@@ -42,66 +43,130 @@ def setUpClass(self):
     def tearDownClass(self):
         pass
 
-    def test_harmonics_sine_wave(self):
-
+    def test_harmonics_sine_wave_pandas(self):
         current = pd.Series(self.signal, index=self.samples)
         harmonics = power.quality.harmonics(current, 1000, self.frequency)
+        
+        for i, j in zip(harmonics['data'].values, self.harmonics_vals):
+            self.assertAlmostEqual(i, j, 1)
+            
+    def test_harmonics_sine_wave_xarray(self):
+        current = xr.DataArray(data=self.signal, 
+                                dims='index',
+                                coords={'index':self.samples})
+        harmonics = power.quality.harmonics(current, 1000, self.frequency)
+        
+        for i, j in zip(harmonics['data'].values, self.harmonics_vals):
+            self.assertAlmostEqual(i, j, 1)
 
-        for i, j in zip(harmonics.values, self.harmonics_vals):
+    def test_harmonic_subgroup_sine_wave_pandas(self):
+        harmonics = pd.DataFrame(self.harmonics_vals,
+                                  index=self.harmonics_int)
+        hsg = power.quality.harmonic_subgroups(harmonics, self.frequency)
+        
+        for i, j in zip(hsg.values, self.harmonic_groups):
             self.assertAlmostEqual(i[0], j, 1)
 
-    def test_harmonic_subgroup_sine_wave(self):
-        current = pd.Series(self.signal, index=self.samples)
-        harmonics = pd.DataFrame(self.harmonics_vals,
-                                 index=self.harmonics_int)
+    def test_harmonic_subgroup_sine_wave_xarray(self):
+        harmonics = xr.Dataset(data_vars={'harmonics':(['index'], self.harmonics_vals)},
+                                coords={'index':self.harmonics_int})
         hsg = power.quality.harmonic_subgroups(harmonics, self.frequency)
+        
         for i, j in zip(hsg.values, self.harmonic_groups):
             self.assertAlmostEqual(i[0], j, 1)
 
-    def test_TCHD_sine_wave(self):
-        current = pd.Series(self.signal, index=self.samples)
+    def test_TCHD_sine_wave_pandas(self):
         harmonics = pd.DataFrame(self.harmonics_vals,
-                                 index=self.harmonics_int)
+                                  index=self.harmonics_int)
         hsg = power.quality.harmonic_subgroups(harmonics, self.frequency)
+        TCHD = power.quality.total_harmonic_current_distortion(
+            hsg)  # had to just put a random rated current in here
```

Comment:
Let's remove all instances of this comment while we are here and document somewhere in set up class or something is its really needed:
> `# had to just put a random rated current in here`

Reply:
Good call. Also i actually just removed the rated current input all together, because it was not being used in anyway. I'm wondering if we're missing a scaling or something for THCD that would use the rated current

Reply:
Reviewing the definition of THCD (not with the marine energy IEC standard), I don't see a rated current used. From what I can tell, THCD is the ratio (in percent) between the energy of the harmonic subgroups and the energy of the dominant frequency and doesn't require a rated current.